src/views/ProductsDetails.vue

```diff
@@ -0,0 +1,212 @@
+          ></v-text-field>
+
+          <resources-dialog
+            v-model="resourceDialog"
+            @save-resources-dialog="resourcesTableValues"
+            @close-dialog="closeDialog"
+          ></resources-dialog>
+
+          <products-dialog v-model="productsDialog" @close-dialog="closeDialog">
+          </products-dialog>
+
+          <div class="d-flex flex-column">
+            <div class="d-flex justify-space-between">
+              <v-btn color="primary" @click="resourceDialog = true">
+                Resources
+              </v-btn>
+              <v-btn color="primary" @click="productsDialog = true">
+                Products
+              </v-btn>
+            </div>
+
+            <v-btn color="success" class="mt-4" block type="submit">
+              Submit
+            </v-btn>
+
+            <v-btn color="error" class="mt-4" block @click="resetForm">
+              Reset
+            </v-btn>
+
+            <v-btn color="warning" class="mt-4" block @click="router.go(-1)">
+              Go Back
+            </v-btn>
+          </div>
+        </v-form>
+      </v-sheet>
+    </v-card>
+  </v-container>
+</template>
+
+<script setup>
+import {
+  useTextFieldLargeRules,
+  useNumberFieldRules,
+  useTextAreaFieldRules,
+} from "@/utils/validation-rules";
+
+import { ref, computed, inject } from "vue";
+import { useRoute, useRouter } from "vue-router";
+import { useStore } from "vuex";
+import ResourcesDialog from "../components/Dialog/ResourcesDialog.vue";
+import ProductsDialog from "@/components/Dialog/ProductsDialog.vue";
+
+const props = defineProps(["VDataTable"]);
+const store = useStore();
+const route = useRoute();
+const router = useRouter();
+const snackbarProvider = inject("snackbarProvider");
+
+const user = computed(() => store.getters["auth/getUser"]).value;
+
+try {
+  await store.dispatch("users/fetchResourcesForUser", user.id);
+} catch (error) {
+  console.log(error);
+}
+
+const [
+  pageTitle,
+  resourceDialog,
+  productsDialog,
+  form,
+  productName,
+  description,
+  authorInput,
+  authors,
+  salePrice,
+  authorsValidation,
+  resourcesContent,
+  productsContent,
+] = [
+  ref(route.meta.title),
+  ref(false),
+  ref(false),
+  ref(null),
+  ref(""),
+  ref(""),
+  ref(""),
+  ref([]),
+  ref(""),
+  ref(false),
+  ref([]),
+  ref(["g28a891r-df13-40eg-9e51-ce313h52f6fi"]),
+];
+
+const addAuthorHandler = () => {
+  if (authorInput.value === "") return;
+  authors.value.push(authorInput.value);
+  authorInput.value = "";
+};
+
+const removeAuthor = (index) => {
+  authors.value.splice(index, 1);
+};
+
+const resetForm = () => {
+  if (form.value) {
+    form.value.reset();
+    form.value.resetValidation();
+    resourcesContent.value = [];
+    authors.value = [];
+    authorsValidation.value = false;
+  }
+};
+
+const closeDialog = (payload) => {
+  payload === "resources"
+    ? (resourceDialog.value = false)
+    : (productsDialog.value = false);
+};
```

Comment:
I would say we create 2 separate methods for closing the different dialogs. Don't think we are gaining anything by reusing this method.